### PR TITLE
Qa 2177 automation ids summary pages

### PR DESCRIPTION
**Pull request by bpierce01**
## Description
- Created automation IDs for following summary pages: project, case, file, gene, mutation, annotation

### CODE REVIEW

File: packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx [hunk 1]
```diff
@@ -59,6 +61,7 @@ export const DownloadFile: React.FC<DownloadFileProps> = ({
   if (file.access === "open") {
     return (
       <DownloadButton
+        data-testid={customDataTestID}
```
Comment (paribartandhakal) [inline packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:64]: Is this part working? Are you able to see the data-testid?

File: packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx [hunk 2]
```diff
@@ -79,6 +82,7 @@ export const DownloadFile: React.FC<DownloadFileProps> = ({
 
   return (
     <DownloadButton
+      data-testid={customDataTestID}
```
Comment (paribartandhakal) [inline packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:85]: Is this part working? Are you able to see the data-testid?

File: packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx (hunk 1 above)
Comment (bpierce01) [reply packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:64]: Yes it appears for open/controlled access file summaries and for the little download file buttons

File: packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx (hunk 2 above)
Comment (bpierce01) [reply packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:85]: yes, see above

**Review by paribartandhakal — APPROVED** [review]

File: packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx (hunk 1 above)
Comment (uc-pperumala) [reply packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:64]: @bpierce01 can you give me an example on how this transforms to for the element?

Comment (uc-pperumala) [reply packages/portal-proto/src/components/DownloadButtons/DownloadFile.tsx:64]: I feel like we would have to set a prefix, such as `button-${customDataTestID}`. Please correct me if I am wrong.

File: packages/portal-proto/src/components/Summary/CategoryTableSummary.tsx [hunk 3]
```diff
@@ -29,12 +31,20 @@ function CategoryTableSummary<TData>({
...
+                size={16}
+                className="text-accent"
+              />
             </ActionIcon>
           </Tooltip>
         )}
       </div>
-      <VerticalTable data={data} columns={columns} />
+      <VerticalTable
+        customDataTestID={customDataTestID}
```
Comment (uc-pperumala) [inline packages/portal-proto/src/components/Summary/CategoryTableSummary.tsx:44]: Same here, will need to set up a prefix such as `text-${customDataTestID}` or something similar?

File: packages/portal-proto/src/components/Table/VerticalTable.tsx [hunk 4]
```diff
@@ -98,6 +99,7 @@ function VerticalTable<TData>({
   setExpanded,
   getRowId = getDefaultRowId,
   baseZIndex = 0,
+  customDataTestID,
```
Comment (uc-pperumala) [inline packages/portal-proto/src/components/Table/VerticalTable.tsx:102]: Just in general, how is the value being set for this variable.

**Review by uc-pperumala — APPROVED** [review]
Just left a few clarifying questions, but otherwise, looks good to me!

File: packages/portal-proto/src/components/Table/VerticalTable.tsx (hunk 4 above)
Comment (bpierce01) [reply packages/portal-proto/src/components/Table/VerticalTable.tsx:102]: Pari could explain more eloquently, but for example VerticalTable is an object that is used throughout the framework. It is a custom object, and we cannot pass a data-testid directly into it. So, we create the variable 'customDataTestID' and then in the object we write data-testid={customDataTestID} at the appropriate location. So then, when we see VerticalTable being used in the framework, we can pass in a customDataTestID at that point and then that gets presented as a data-testid.

File: packages/portal-proto/src/components/Summary/CategoryTableSummary.tsx (hunk 3 above)
Comment (bpierce01) [reply packages/portal-proto/src/components/Summary/CategoryTableSummary.tsx:44]: This one was more complex where I needed to pass in a 'customDataTestID' through two layers before it shows up as the data-testid

**Review by craigrbarnes — APPROVED** [review]
Changes are fine. Tested tables and other components. updated in this PR, and see no issues

**Pull request merged**